compaction/compaction.cc

```diff
@@ -1125,7 +1125,7 @@ class cleanup_compaction final : public regular_compaction {
         : cleanup_compaction(table_s, std::move(descriptor), cdata, opts.owned_ranges) {}
 
     flat_mutation_reader make_sstable_reader() const override {
-        return make_filtering_reader(regular_compaction::make_sstable_reader(), make_partition_filter());
+        return downgrade_to_v1(make_filtering_reader(upgrade_to_v2(regular_compaction::make_sstable_reader()), make_partition_filter()));
```

Comment:
this is somewhat sad. couldn't flat_mutation_reader become a base class instead? or we intend to eventually merge v2 into v1 and deprecate the latter?

Reply:
When everything is converted to v2, these conversions will disappear.